core/src/main/java/com/amplifyframework/devmenu/EnvironmentInfo.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.devmenu;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.Resources;
+import com.amplifyframework.core.category.Category;
+import com.amplifyframework.core.plugin.Plugin;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Contains information about the plugin versions added to the application
+ * and the developer's environment.
+ */
+public final class EnvironmentInfo {
+    // Name of the file that contains the developer environment information.
+    private static final String DEV_ENV_INFO_FILE_NAME = "localenvinfo";
+    // A map from developer environment information items to the formatted name of that item.
+    private final Map<String, String> devEnvironmentItems;
+
+    /**
+     * Constructs a new EnvironmentInfo instance.
+     */
+    public EnvironmentInfo() {
+        devEnvironmentItems = new HashMap<>();
+        devEnvironmentItems.put("nodejsVersion", "Node.js Version");
+        devEnvironmentItems.put("npmVersion", "NPM Version");
+        devEnvironmentItems.put("amplifyCliVersion", "Amplify CLI Version");
+        devEnvironmentItems.put("androidStudioVersion", "Android Studio Version");
+        devEnvironmentItems.put("osName", "OS");
+        devEnvironmentItems.put("osVersion", "OS Version");
+    }
+
+    /**
+     * Returns a String representation of the version of each plugin added, or
+     * "No plugins added." if no plugins were added.
+     * @return version information for each plugin added
+     */
+    public String getPluginVersions() {
+        String pluginVersions = getCategoryPluginVersions(Amplify.Analytics) +
+                getCategoryPluginVersions(Amplify.API) +
+                getCategoryPluginVersions(Amplify.Auth) +
+                getCategoryPluginVersions(Amplify.Logging) +
+                getCategoryPluginVersions(Amplify.Storage) +
+                getCategoryPluginVersions(Amplify.Hub) +
+                getCategoryPluginVersions(Amplify.DataStore) +
+                getCategoryPluginVersions(Amplify.Predictions);
```

Comment:
In situations like this, it's best to separate out a collection, and then use control flow. e.g.,

```java
// TODO: can we centralize these, somewhere? We want ~1 file in the code base that lists all the categories.
List<Category<?>> categories = Arrays.asList(Amplify.Whatever, ...);
for (Category<?> c : categories) {
    getCategoryPluginVersions(c);
}
```